opsramp/resources.py

```diff
@@ -89,3 +89,9 @@ def availability(self, uuid, start_epoch, end_epoch):
             uuid, start_epoch, end_epoch
         )
         return self.api.get(url_suffix)
+
+    def get_templates(self, uuid, pattern=''):
+        url_suffix = '{0}/templates/search?{1}'.format(
+            uuid, pattern
+        )
+        return self.api.get(url_suffix)
```

Comment:
Calling api.get will work as long as there is only one page of results. However I suggest changing to the following, which takes advantage of the superclass's support for paginated search results.
```
def get_templates(self, uuid, pattern=None):
     url_suffix = '{0}/templates'.format(uuid)
     self.api.pushd(url_suffix)
     resp = super(Resources, self).search(pattern)
     resp = self.search(pattern)
     self.api.popd()
     return resp
```

Reply:
thanks Josie. Fixed it